src/core/error/common_error_map.rs

```diff
@@ -0,0 +1,56 @@
+use de::StdError;
+
+use crate::prelude::*;
+
+pub trait MapToFailedToDeserializeJSONToValue<R> {
+    fn map_failed_to_deserialize_string<T>(
+        self,
+        input: impl AsRef<str>,
+    ) -> Result<R, CommonError>;
+    fn map_failed_to_deserialize_bytes<T>(
+        self,
+        input: &[u8],
+    ) -> Result<R, CommonError>;
+}
+
+impl<R> MapToFailedToDeserializeJSONToValue<R>
+    for Result<R, serde_json::Error>
+{
+    fn map_failed_to_deserialize_string<T>(
+        self,
+        input: impl AsRef<str>,
+    ) -> Result<R, CommonError> {
+        self.map_err(|e| {
+            error!(
+                "Failed to deserialize JSON to {}, from:\n{:?}\nError: {}",
+                type_name::<T>(),
+                input.as_ref(),
+                e
+            );
+            CommonError::FailedToDeserializeJSONToValue {
+                json_byte_count: input.as_ref().len() as u64,
+                type_name: type_name::<T>(),
+                message: format!("{}", e),
```

Comment:
if you do `format!("{:?}", e)` instead, do we get any more meaningful info? i.e. does `"expected value at line 1 column 1"` change into something even more helpful?

Reply:
Not particularly. For this mentioned example we get `Error(\"expected value\", line: 1, column: 1)`

Reply:
mm not so helpful :P